src/findnacld.c

```diff
@@ -0,0 +1,193 @@
+/* Copyright (c) 2015 The crouton Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+#include "websocket.h"
+#include <unistd.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <sys/epoll.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <sys/select.h>
+#include <stddef.h>
+
+#define MAX_EVENTS 100
+
+const char *SOCKET_DIR = "/var/run/crouton-ext";
+const char *SOCKET_PATH = "/var/run/crouton-ext/socket";
+
+int send_pid_fd(int conn, long pid, int fd)
+{
+    struct msghdr msg = {0};
+    struct cmsghdr *cmsg;
+    struct iovec iov;
+    char buf[CMSG_SPACE(sizeof(int))]; /* ancillary data buffer */
+
+    /* It is not necessary to sent the pid. However, to pass fd using sendmsg,
+     * at least 1 byte of data must be sent.
+     */
+    iov.iov_base = &pid;
+    iov.iov_len = sizeof(pid);
+
+    msg.msg_iov = &iov;
+    msg.msg_iovlen = 1;
+    if (fd > 0) {
+        msg.msg_control = buf;
+        msg.msg_controllen = CMSG_SPACE(sizeof(int));
+
+        cmsg = CMSG_FIRSTHDR(&msg);
+        cmsg->cmsg_level = SOL_SOCKET;
+        cmsg->cmsg_type = SCM_RIGHTS;
+        cmsg->cmsg_len = CMSG_LEN(sizeof(int));
+
+        *((int *)CMSG_DATA(cmsg)) = fd;
+    } else {
+        msg.msg_control = NULL;
+        msg.msg_controllen = 0;
+    }
+
+    return sendmsg(conn, &msg, 0);
+}
+
+
+int find_nacl(int conn)
+{
+    char argbuf[70], outbuf[256];
+    char* cut;
+    int idx = 0, c, len;
+
+    if ((len = read(conn, argbuf, sizeof(argbuf)-1)) < 0) {
+       syserror("Failed to read arguments");
+       return -1;
+    }
+    argbuf[len] = 0;
+
+    cut = strchr(argbuf, ' ');
+    if (!cut) {
+        error("No ' ' in findnacl arguments: %s.", argbuf);
+        return -1;
+    }
+    *cut = 0;
+
+    char *cmd = "croutonfindnacl";
+    char* args[] = {cmd, argbuf, cut + 1, NULL};
+
+    c = popen2(cmd, args, NULL, 0, outbuf, sizeof(outbuf)-1);
```

Comment:
Because this is from fbserver.c. To consist with fbserver.c, I've changed `len` to `c`.